sql/changes/1.12/add-stored-order-quote-taxes.sql

```diff
@@ -0,0 +1,62 @@
+
+
+create table oe_tax (
+  oe_id int not null references oe (id),
+  tax_id int not null references account (id),
+  tax_basis numeric not null,
+  exempt smallint not null default 0,
+  rate numeric,
+  tax_amount numeric,
+  source text,
+  primary key (oe_id, tax_id),
+  check (exempt = 0 or (rate is null and tax_amount is null)),
+  check (exempt <> 0 or (rate is not null and tax_amount is not null))
+);
+
+create table tax_exempt_reason (
+  id serial not null primary key,
```

Comment:
I chose to stick with this way because the "rest" of the schema also uses it. I'd say that if we want to migrate to the `generated` form (yes, I'm all for that!), we'd probably be better off moving the entire schema over instead of just starting to use the new style "somewhere".

Reply:
> I chose to stick with this way because the "rest" of the schema also uses it. I'd say that if we want to migrate to the `generated` form (yes, I'm all for that!), we'd probably be better off moving the entire schema over instead of just starting to use the new style "somewhere".

I understand, but if we do it the better way every time we create or update a table definition, then the code base gradually gets better.  If we wait, it will never get better because the time is better spent elsewhere.

Reply:
We agreed that @neilt will create a PR changing all "serial" columns to `int generated`. Awaiting that for now.

Reply:
With the upcoming PR to replace `serial` with `generated [...] identity`, create the new table that way.